src/analysis/interpolation/qgsidwinterpolator.h

```diff
@@ -60,11 +65,38 @@ class ANALYSIS_EXPORT QgsIDWInterpolator: public QgsInterpolator
     */
     double distanceCoefficient() const { return mDistanceCoefficient; }
 
+    double interpolatedPoint( const QgsPointXY &point, QgsFeedback *feedback ) const override;
+
   private:
 
     QgsIDWInterpolator() = delete;
 
     double mDistanceCoefficient = 2.0;
+
+    //! Cached vertex data for input sources
+    QVector<QgsInterpolatorVertexData> mCachedBaseData;
```

Comment:
What if the source layer is huge? Won't this limit the side of layer you can interpolate from?

Reply:
Using std::vector now